srv/components/provisioner/files/glusterfshsaredstorage.service

```diff
@@ -1,9 +1,7 @@
 [Unit]
 Description=Mount glusterfs sharedstorage
-After=glusterd.service
-Before=salt-minion.service
-Requires=glusterd.service salt-minion.service remote-fs-pre.target local-fs.target
-BindsTo=glusterd.service salt-minion.service
+Before=salt-master.service
```

Comment:
@andkononykhin2 
We have observed
If we add dependency as Wants/Requires in [Unit] section, glusterfssharedstorage service will make sure salt-master is started.
but if we start salt-master service, glusterfssharedstorage service will not be started and mounts will not be present internally

hence we decided to use Wantedby/RequiredBy in [Install] section
here on salt-master start it will make sure the glusterfssharedstorage and glusterd services are started

Reply:
@sradhanand-pati , @andkononykhin2 
I'd have a short discussion with @SomwanshiAnjali. In my opinion, it would be better to move the wants section and after requirement to master service file and not modify any gluster service files.

Reason:
1. I was adamant on gluster service file initially as I wanted the changes to have minimal impacts by 3rd Dec.
1. It would be nice to move to dependencies to [Unit] section as executing `systemctl enable` midway is not a good idea
1. There is a need to upgrade all softwares in near future, so we would need to handle this file update post update anyways, so there's particular reason for selecting glusterfs over salt-master

Do share your thoughts.

Reply:
@ypise @sradhanand-pati @SomwanshiAnjali  
My preference is [Unit] section too. But looks like (docs are not so clear on that) it WantedBy= doesn't work as I expected. So I would suggest:
- merge the PR in the current state
- create a separate PR (task)  to move that relation into salt-master as it was mentioned here